lib/button.js

```diff
@@ -0,0 +1,326 @@
+      }
+    });
+    button.addEventListener('mouseout', function () {
+      if (!self.buttonMenu) {
+        self.createInlineStyle(button, 'canvas-datagrid-button-wrapper');
+      }
+    });
+  }
+
+  function createButton(pos, items, imgSrc) {
+    var wrapper = document.createElement('div'),
+      buttonArrow = document.createElement('div'),
+      buttonIcon = document.createElement('div'),
+      intf = {};
+
+    if (!Array.isArray(items)) {
+      throw new Error('createButton expects an array.');
+    }
+
+    function init() {
+      var loc = {},
+        s = self.scrollOffset(self.canvas);
+
+      if (zIndexTop === undefined) {
+        zIndexTop = self.style.buttonZIndex;
+      }
+
+      applyButtonStyle(wrapper);
+      self.createInlineStyle(buttonIcon, 'canvas-datagrid-button-icon');
+      self.createInlineStyle(buttonArrow, 'canvas-datagrid-button-arrow');
+
+      loc.x = pos.left - s.left;
+      loc.y = pos.top - s.top;
+      loc.height = 0;
+      zIndexTop += 1;
+      wrapper.style.position = 'absolute';
+      wrapper.style.zIndex = zIndexTop;
+      wrapper.style.left = loc.x + 'px';
+      wrapper.style.top = loc.y + 'px';
+      wrapper.left = pos.left + self.scrollBox.scrollLeft;
+      wrapper.top = pos.top + self.scrollBox.scrollTop;
+      buttonArrow.innerHTML = self.style.buttonArrowDownHTML;
+      if (imgSrc) {
+        let img = document.createElement('img');
+        img.setAttribute('src', imgSrc);
+        img.style.maxWidth = '100%';
+        img.style.height = '100%';
+        buttonIcon.appendChild(img);
+      }
+      wrapper.appendChild(buttonIcon);
+      wrapper.appendChild(buttonArrow);
+      document.body.appendChild(wrapper);
+      wrapper.addEventListener('click', toogleButtonMenu);
+    }
+
+    intf.wrapper = wrapper;
+    intf.items = items;
+    init();
+
+    intf.dispose = function () {
+      if (wrapper.parentNode) {
+        wrapper.parentNode.removeChild(wrapper);
+      }
+    };
+    return intf;
+  }
+
+  function toogleButtonMenu() {
+    function createDisposeEvent() {
+      requestAnimationFrame(function () {
+        document.addEventListener('click', self.disposeButtonMenu);
+      });
+    }
+    if (self.buttonMenu) {
+      self.disposeButtonMenu();
+    } else {
+      let pos = {
+        left: self.button.wrapper.left - self.scrollBox.scrollLeft,
+        top:
+          self.button.wrapper.top +
+          self.button.wrapper.offsetHeight -
+          self.scrollBox.scrollTop,
+      };
+      self.buttonMenu = createButtonMenu(pos, self.button.items);
+      self.createInlineStyle(
+        self.button.wrapper,
+        'canvas-datagrid-button-wrapper:active',
+      );
+      createDisposeEvent();
+    }
+  }
+
+  function createButtonMenu(pos, items) {
+    var buttonMenu = document.createElement('div'),
+      selectedIndex = -1,
+      intf = {},
+      rect;
+
+    function createItems() {
+      items.forEach(function (item) {
```

Comment:
For all applicable occurrences in this PR (where it makes sense, that is), would it be possible to replace `.forEach` with a for-of loop? For example,

```js
for (const el of array) {
  // do something
}
```

They are [faster than forEach](https://coderwall.com/p/kvzbpa/don-t-use-array-foreach-use-for-instead), and I find the syntactic overhead of a function distracting and unnecessary.